src/cholesky.jl

```diff
@@ -1,21 +1,30 @@
 # Generic Cholesky decomposition for fixed-size matrices, mostly unrolled
 non_hermitian_error() = throw(LinearAlgebra.PosDefException(-1))
-@inline function LinearAlgebra.cholesky(A::StaticMatrix)
+@inline function LinearAlgebra.cholesky(A::StaticMatrix; check::Bool = true)
     ishermitian(A) || non_hermitian_error()
-    C = _cholesky(Size(A), A)
-    return Cholesky(C, 'U', 0)
+    _cholesky(Size(A), A, check)
 end
 
-@inline function LinearAlgebra.cholesky(A::LinearAlgebra.RealHermSymComplexHerm{<:Real, <:StaticMatrix})
-    C = _cholesky(Size(A), A.data)
-    return Cholesky(C, 'U', 0)
+@inline function LinearAlgebra.cholesky(A::LinearAlgebra.RealHermSymComplexHerm{<:Real, <:StaticMatrix}; check::Bool = true)
+    _cholesky(Size(A), A.data, check)
 end
 @inline LinearAlgebra._chol!(A::StaticMatrix, ::Type{UpperTriangular}) = (cholesky(A).U, 0)
 
-@generated function _cholesky(::Size{S}, A::StaticMatrix{M,M}) where {S,M}
+@inline function _chol_failure(A, info, check)
+    if check
+        throw(LinearAlgebra.PosDefException(info))
+    else
+        Cholesky(A, 'U', info)
+    end
+end
+# x < zero(x) is check used in `sqrt`, letting LLVM eliminate that check and remove error code.
+@inline _nonpdcheck(x::Real) = x ≥ zero(x)
+@inline _nonpdcheck(x) = x == x
+
+@generated function _cholesky(::Size{S}, A::StaticMatrix{M,M}, check::Bool) where {S,M}
     @assert (M,M) == S
-    M > 24 && return :(_cholesky_large(Size{$S}(), A))
-    q = Expr(:block)
+    M > 24 && return :(_cholesky_large(Size{$S}(), A, check))
+    q = Expr(:block, :(info = 0), :(failure = false))
```

Comment:
Are `info` and `failure` used somewhere? I can't see that.

Reply:
Good catch. Leftovers from an earlier attempt that took >10ns for the 4x4 example.